tasks/Logger.cpp

```diff
@@ -411,9 +388,102 @@ bool Logger::handleExistingFile()
         do {
             log(Warning) << "Timestamped file " << _current_file.get() << " already exists. Retrying in 1 second." << endlog();
             usleep(1*1000000);
-            timestampFile();
+            timestampFile(value);
         } while(boost::filesystem::exists(_current_file.get()));
         log(Warning) << "Writing log to " << _current_file.get() << " instead." << endlog();
         return true;
     }
 }
+
+bool Logger::setCurrentFile(::std::string const& value)
+{
+    if(value.empty())
+    {
+      log(Error) << "Could not create log file. Task property _file is empty." << endlog();
+      return false;
+    }
+
+    if (_overwrite_existing_files.get() && _auto_timestamp_files.get())
+    {
+        log(Error) << "The properties overwrite_existing_files and auto_timestamp_files are both set to true, but are mutually exclusive." << endlog();
+        return false;
+    }
+
+    _current_file.set(value);
+
+    if (_auto_timestamp_files.get()) { timestampFile(value); }
+
+    if (boost::filesystem::exists(_current_file.get()))
+    {
+        if (!handleExistingFile(value))
+        {
+            return false;
+        }
+    } else if (_auto_timestamp_files.get())
+    {
+        log(Info) << "Successfully timestamped log file. Writing to: " << _current_file.get() << endlog();
+    }
+    return true;
+}
+
+void Logger::updateLoggers()
```

Comment:
This is logic that's very close to the one in `startHook` I'd call it there. The main change I see is that you'd need to guard the `m_io->close()` with null check